iothub_client/src/iothub_client_core_ll.c

```diff
@@ -1743,11 +1745,29 @@ void IoTHubClientCore_LL_Destroy(IOTHUB_CLIENT_CORE_LL_HANDLE iotHubClientHandle
         while ((unsend = DList_RemoveHeadList(&(handleData->iot_msg_queue))) != &(handleData->iot_msg_queue))
         {
             IOTHUB_DEVICE_TWIN* temp = containingRecord(unsend, IOTHUB_DEVICE_TWIN, entry);
+
+            // The Twin reported properties status codes are based on HTTP codes and provided by the service.
+            // Following design already implemented in the transport layer, the status code shall be artificially 
+            // returned as zero to indicate the report was not sent due to the client being destroyed.
+            if (temp->reported_state_callback != NULL)
+            {
+                temp->reported_state_callback(ERROR_CODE_BECAUSE_DESTROY, temp->context);
```

Comment:
Line 1738 above has `IOTHUB_CLIENT_CONFIRMATION_BECAUSE_DESTROY`. Is that a different error code?

Reply:
That is for telemetry.
The callback definitions are different. Twin has a concept of a status code that comes from the service as a response to the reported state update request.

Reply:
Gotcha thanks